content/systems/filecoin_vm/_index.md

```diff
@@ -4,12 +4,43 @@ description: VM - Virtual Machine
 bookCollapseSection: true
 weight: 3
 dashboardWeight: 2
-dashboardState: wip
+dashboardState: incomplete
 dashboardAudit: 0
 dashboardTests: 0
 ---
 
 # VM - Virtual Machine
 ---
 
-{{<embed src="vm.id" lang="go" >}}
+An Actor in the Filecoin Blockchain is the equivalent of the smart contract in the Ethereum Virtual Machine. Actors carry the logic needed in order to submit transactions, proofs and blocks, among other things, to the Filecoin blockchain. Every actor is identified by a unique address.
+
+The Filecoin Virtual Machine (VM) is the system component that is in charge of execution of all actors code. Execution of actors on the Filecoin VM (i.e., on-chain executions) incur a gas cost.
+
+Any operation applied (i.e., executed) on the Filecoin VM produces an output in the form of a _State Tree_ (discussed below). The latest _State Tree_ is the current source of truth in the Filecoin Blockchain. The _State Tree_ is identified by a CID, which is stored in the IPLD store.
+
+```go
+type VM struct {
+	## The current State Tree
+	cstate      *state.StateTree
+	
+	## The CID (i.e., identifier) of the current State Tree
+	base        cid.Cid
+	
+	## The IPLD store of the CID (of the current State Tree)
+	cst         *cbor.BasicIpldStore
+	
+	## Current BlockStore
+	buf         *bufbstore.BufferedBS
+	
+	## The epoch number where the VM has been called
+	blockHeight abi.ChainEpoch
+	
+	## The method invoking a State Tree change
+	inv         *Invoker
+	
+	## Randomness included in messages submitted to the VM
+	rand        Rand
+
+	Syscalls runtime.Syscalls
```

Comment:
I'm not sure what to say about code snippet here. This is invoking a bunch of concepts like `Invoker` and `StateTree` that aren't (AFAIK) defined and mix in a bunch of implementation-specific details that depend on where the code was lifted from. So it has a bunch of low-level, un-explained details, but not enough to be a complete implementation.

Do you want me to critique this down to a minimal pseudo-Go structure?